subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/CacheBackedTaskHistoryRepository.java

```diff
@@ -361,7 +350,7 @@ private static OverlappingOutputs detectOverlappingOutputs(ImmutableSortedMap<St
         return null;
     }
 
-    private static FileCollectionSnapshot getSnapshotAfterPreviousExecution(TaskExecution previousExecution, String propertyName) {
+    private static FileCollectionSnapshot getSnapshotAfterPreviousExecution(@Nullable AbstractTaskExecution previousExecution, String propertyName) {
```

Comment:
If this takes a previous execution, then the type should be `HistoricTaskExecution`. If not, then the method name should be `getSnapshotAfterExecution`.

Reply:
Good catch! I removed all usages of `AbstractTaskExecution`.